### PR TITLE
Merge in ES6 and test cases

### PR DESCRIPTION
**Pull request by taitung**
Refactor to follow ES6 and add  test cases

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  app.use(async (ctx, next) => {
+    try {
+      await next();
+    } catch (err) {
+      debug('error caught!', err.message);
+      // refer to https://stackoverflow.com/questions/18391212/is-it-not-possible-to-stringify-an-error-using-json-stringify
+      // TODO: will need to follow up the order of the stack coming from the key fetching from
+      //       getOwnPropertyNames
+      let errMsg = '';
+      Object.getOwnPropertyNames(err).forEach(key => {
```
Comment (AllenFang) [inline src/index.js]: Not sure if we can use `reduce` here: LOL
```js
const errMsg = Object.getOwnPropertyNames(err).reduce((prev, curr) => `${prev} ${curr} ${err[curr]}\r\n`);
```

File: src/example/store.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const { id } = ctx.params;
+  if (!id) throw new TypeError('id is not found!');
+  const obj = ctx.request.body;
+  if (!obj || !Object.keys(obj).length) {
+    ctx.status = 400;
+    return;
+  }
+  const { fields } = ctx.request.body;
+  const { field } = fields[0];
+  const { value } = fields[0];
```
Comment (AllenFang) [inline src/example/store.js]: `const [field, value] = fields;` will be more simple

File: src/example/store.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  updatedStore.address = store.address;
+  stores.taiwan[0] = updatedStore;
+  ctx.status = 200;
+  ctx.type = 'application/json';
+  ctx.body = stores;
+};
+
+export const remove = ctx => {
+  const { id } = ctx.params;
+  if (!id) throw new TypeError('id is not found!');
```
[the remark this answers is not in the document]
Comment (taitung) [reply src/example/store.js]: will remove this line

File: package.json [hunk 4]
```diff
@@ -1,22 +1,55 @@
...
-    "koa": "^2.4.1"
+    "babel-core": "6.26.0",
+    "commander": "2.12.2",
+    "debug": "3.1.0",
+    "del": "3.0.0",
+    "global": "4.3.2",
+    "gulp": "3.9.1",
+    "gulp-babel": "7.0.0",
+    "gulp-eslint": "4.0.0",
+    "gulp-mocha": "4.3.1",
```
[the remark this answers is not in the document]
Comment (taitung) [reply package.json]: Gulpfile.js right now is scanned by eslint. I will put this eslintignore

File: src/example/store.js (hunk 2 above)
Comment (taitung) [reply src/example/store.js]: i see

File: src/index.js (hunk 1 above)
Comment (taitung) [reply src/index.js]: nice!

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      const { handler, routes } = binding;
+      const modPath = join(__dirname, handler);
+      if (!fs.existsSync(`${modPath}.js`)) {
+        throw new TypeError(`${modPath} is not found!`);
+      }
+      /* eslint-disable global-require, import/no-dynamic-require */
+      const module = require(modPath);
+      Object.keys(routes).forEach(route => {
+        const method = route.split(' ')[0];
+        const subUrl = route.split(' ')[1];
```
[the remark this answers is not in the document]
Comment (taitung) [reply src/index.js]: got it

File: test.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
+/**
+ * Dependencies
+ */
+const chaiHttp = require('chai-http');
+const mocha = require('mocha');
+const chai = require('chai');
+
+// local require
+const sbkoaapis = require('./index.js');
```
[the remark this answers is not in the document]
Comment (taitung) [reply test.js]: ./

**Comment by taitung** [comment]
@AllenFang just pushed the modification. please help review 😀

File: package.json [hunk 7]
```diff
@@ -1,22 +1,55 @@
...
+  "version": "1.0.0",
+  "description": "A RESTful APIs skeleton codes based on Koa",
+  "main": "bin/run",
   "dependencies": {
-    "koa": "^2.4.1"
+    "babel-core": "6.26.0",
+    "commander": "2.12.2",
+    "debug": "3.1.0",
+    "del": "3.0.0",
+    "global": "^4.3.2",
```
Comment (AllenFang) [inline package.json]: Why need this? where we used?

Comment (taitung) [reply package.json]: ah... the order is important! `yarn add global ...` I will remove it

**Comment by taitung** [comment]
@AllenFang thanks! I removed Global and merge it

**Pull request merged**